src/app/index.js

```diff
@@ -1,26 +1,40 @@
-import React from "react";
+import React, { useState } from "react";
 import ReactDOM from "react-dom";
 import App from "./App";
 import { configureStore } from "@reduxjs/toolkit";
 import rootReducer from "./reducers";
 import { Provider } from "react-redux";
 import "./styles.css";
+import { fakeReduxStore } from "../fake-redux-store";
 
 const store = configureStore({ reducer: rootReducer });
 
 function AppContainer() {
+  const [fake, setFake] = useState(false);
+  const StoreProvider = fake ? FakeStoreProvider : RealStoreProvider;
```

Comment:
@dtinth So in Taskworld application, we will use `process.env.NODE_ENV === 'test'` to indicate that we should use fake redux store right?

Reply:
@NesChaiyapon Nope, since the entrypoint is different for our app and our test, we don't need conditional logic, as the main app will always use the RealStoreProvider, whereas in tests and storybooks we always use FakeStoreProvider. ;)

Reply:
@dtinth  I see. Do you mean that a Fake and a Real redux store will be automatically selected in our app because the entrypoint is different for our app and our test? Can I ask you one question? What is an entrypoint? 🤣

Reply:
What I mean, in our app, it will look like this:

```js
ReactDOM.render(
  <RealStoreProvider>
    <App />
  </RealStoreProvider>,
  element
)
```

whereas in our tests, it will look like this:

```js
const { container } = render(
  <FakeStoreProvider>
    <MyComponent />
  </FakeStoreProvider>
)
```

as well as in our storybook, our `<Wrapper>` component in `.storybook/config.js` will put the component into `<FakeStoreProvider>` automatically.

> Can I ask you one question? What is an entrypoint? 🤣

This is where code gets executed. For apps created with `create-react-app` the entrypoint to the app is `src/index.js`. When running tests, each test file becomes an entrypoint at which Jest begins executing tests (which is each `.test.js` file). For storybook, the entrypoint is `.storybook/config.js`.